### PR TITLE
Connection: Move the Jetpack IXR client to the package

### PR DESCRIPTION
**Pull request by tyxla**
In our quest to make the connection package fully independent, we still have to decouple some bits of it that belong to the Jetpack core. Some of those parts belong to other packages too, and some of them even belong to the connection package.

One of those pieces is the Jetpack IXR client. It is currently in the core of Jetpack, so we have to decide what to do with it. Since it relies on the connection package, and the connection package relies on it, that means that it makes sense to be part of the connection package.

So, this PR moves the Jetpack IXR client to the connection package.

#### Changes proposed in this Pull Request:
* Move `Jetpack_IXR_Client` to the connection package (classmapped).
* Move `Jetpack_IXR_ClientMulticall` to the connection package (classmapped).
* Introduce a basic `IXR` loader to load the WP core `IXR_` classes that we extend.
* Autoload the `IXR` loader by using the `files` Composer autoload directive.
* Deprecate `class.jetpack-ixr-client.php`.
* Deprecate `Jetpack::load_xml_rpc_client()`.
* Remove any unnecessary usage of `Jetpack::load_xml_rpc_client()`.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This is a continuation of this effort: p1HpG7-7lI-p2

#### Testing instructions:
* 😅
* Checkout this branch.
* Run `composer install`
* Enable `WP_DEBUG` and `WP_DEBUG_LOG`.
* Test a bunch of things, like:
  * Connecting/reconnecting Jetpack to WP.com.
  * Disconnecting Jetpack from WP.com
  * Saving some Jetpack settings in wp-admin (toggles for example)
  * Saving some Jetpack settings in Calypso (toggles for example)
  * Incremental sync - trigger an incremental sync by creating a new draft post
  * Full sync - trigger a full sync from the debugger
  * Plugin/theme autoupdate
  * Optional: some Jetpack modules
    * Monitor - enable it and try breaking your site
    * Protect - enable it and try requesting the site from a forbidden IP
    * Post by email - try posting by email
    * Publicize - try it with a connected service and verify it publicized properly on it
    * Subscriptions - verify emails for new posts are still sent
    * SSO - try logging in with SSO.
* Verify there are no errors logged and everything worked well.
* Verify all tests still pass

#### Proposed changelog entry for your changes:
* Move `Jetpack_IXR_Client` to the connection package (classmapped).
* Move `Jetpack_IXR_ClientMulticall` to the connection package (classmapped).
* Introduce a basic `IXR` loader to load the WP core `IXR_` classes that we extend.
* Autoload the `IXR` loader by using the `files` Composer autoload directive.
* Deprecate `class.jetpack-ixr-client.php`.
* Deprecate `Jetpack::load_xml_rpc_client()`.
* Remove any unnecessary usage of `Jetpack::load_xml_rpc_client()`.

### CODE REVIEW

**Review by roccotripaldi — DISMISSED** [review]
I tested most of the flows you suggested in the testing instructions, and I'm confident that the things are still running smoothly. 

I only noticed one error, which I addressed in dd85c2094b6f7c4e70444540b04a09568f990205

**Comment by roccotripaldi** [comment]
And my change made the tests fail :(
I will investigate.

File: class.jetpack.php [hunk 1]
```diff
@@ -574,8 +574,7 @@ private function __construct() {
 			};
 		} );
 
-		$this->connection_manager = new Connection_Manager();
-		$this->connection_manager->init();
+		add_action( 'plugins_loaded', array( $this, 'init_connection_manager' ) );
```
Comment (tyxla) [inline class.jetpack.php]: @roccotripaldi I'm not sure I agree with this change. There are a lot of pieces in the code that rely that `Jetpack::connection()` (respectively `Jetpack::init()`) will grant us with a connection manager instance that we can use immediately. With this change, we will no longer be able to confirm this assumption, and some places will start trying to access an object that's not constructed yet.

What was the motivation for this change? Could we figure out any alternatives?

Comment (tyxla) [reply class.jetpack.php]: Reverted it, and filed a fix for the bug Rocco discovered in https://github.com/Automattic/jetpack/pull/13270/commits/43722205570a79e7c30b66c7dd3bf7552d739cde.

**Comment by tyxla** [comment]
Ready for another review!

**Review by roccotripaldi — APPROVED** [review]
I'm still seeing errors in my debug.log like `PHP Notice:  Jetpack_XMLRPC_Server was called <strong>incorrectly</strong>.`

After a call with @tyxla this morning, we traced the cause to some plugins i've had installed:

- WooCommerce
- WooCommerce services

I approve this PR and we can address the Woo conflicts in the next PR.

**Comment by tyxla** [comment]
Thanks, going to look at those Woo errors soon!

**Comment by jeherve** [comment]
r206808-wpcom

**Pull request merged**